cosmwasm/src/encoding.rs

```diff
@@ -0,0 +1,86 @@
+use crate::types::{
+    ActionType, ReinitResponseData, StakeResponseData, Status, UnstakeResponseData,
+};
+use cosmwasm_std::{Binary, Uint128};
+
+pub fn encode_stake_payload(stake_id: &u64) -> Binary {
+    let payload: Vec<u8> = stake_id
+        .to_be_bytes()
+        .into_iter()
+        .chain(vec![0_u8].into_iter()) // ActionType: stake = 0_u8
```

Comment:
```suggestion
        .chain(Some(0)) // ActionType: stake = 0_u8
```
I think it's better because no need for allocation 1 element vector